easybuild/easyconfigs/r/R/R-3.4.3-foss-2017b-X11-20171023.eb

```diff
@@ -0,0 +1,3459 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+# Author: Jonas Demeulemeester
+# The Francis Crick Institute
+
+name = 'R'
+version = '3.4.3'
+x11ver = '20171023'
+versionsuffix = '-X11-%s' % x11ver
+
+homepage = 'http://www.r-project.org/'
+description = """R is a free software environment for statistical computing and graphics."""
+
+toolchain = {'name': 'foss', 'version': '2017b'}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://cran.us.r-project.org/src/base/R-%(version_major)s']
+
+checksums = ['7a3cb831de5b4151e1f890113ed207527b7d4b16df9ec6b35e0964170007f426']
+
+configopts = "--with-pic --enable-threads --enable-R-shlib"
+# some recommended packages may fail in a parallel build (e.g. Matrix), and we're installing them anyway below
+configopts += " --with-recommended-packages=no"
+
+dependencies = [
+    ('X11', x11ver),
+    ('Mesa', '17.2.5'),
+    ('libGLU', '9.0.0'),
+    ('cairo', '1.14.10'),
+    ('libreadline', '7.0'),
+    ('ncurses', '6.0'),
+    ('bzip2', '1.0.6'),
+    ('XZ', '5.2.3'),
+    ('zlib', '1.2.11'),
+    ('SQLite', '3.20.1'),
+    ('PCRE', '8.41'),
+    ('libpng', '1.6.32'),  # for plotting in R
+    ('libjpeg-turbo', '1.5.2'),  # for plottting in R
+    ('LibTIFF', '4.0.9'),
+    ('Java', '1.8.0_152', '', True),  # Java bindings are built if Java is found, might as well provide it
+    ('Tcl', '8.6.7'),  # for tcltk
+    ('Tk', '8.6.7'),  # for tcltk
+    ('cURL', '7.55.1'),  # for RCurl
+    ('libxml2', '2.9.4'),  # for XML
+    ('GDAL', '2.2.3', '-Python-2.7.14'),  # for rgdal
+    ('PROJ', '4.9.3'),  # for rgdal
+    ('GMP', '6.1.2'),  # for igraph
+    ('NLopt', '2.4.2'),  # for nloptr
+    ('FFTW', '3.3.6'),  # for fftw
+    ('libsndfile', '1.0.28'), # for seewave
+    # OS dependency should be preferred if the os version is more recent then this version,
+    # it's nice to have an up to date openssl for security reasons
+    # ('OpenSSL', '1.0.2h'),
+]
+
+osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
+
+name_tmpl = '%(name)s_%(version)s.tar.gz'
+ext_options = {
```

Comment:
Please use the new `exts_default_options` construct added in https://github.com/easybuilders/easybuild-framework/pull/2345, see also easyconfig for R 3.4.3 that just got merged.

The easiest way to tackle this is probably to copy the contents from https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/r/R/R-3.4.3-intel-2017b-X11-20171023.eb, and change the toolchain to `foss`. By preference, use the same dependency versions (it sort of useful to keep easyconfigs consistent between the `foss` and `intel` common toolchains).

Reply:
I tried to copy your easyconfig file for intel2017b, but this is not building because it tries do download the extensions from http://cran.us.r-project.org/src/base/ instead of http://cran.r-project.org/src/contrib/. I have no idea, why.

Reply:
@Helios07 If you're not using current `develop` of EasyBuild (which will become v3.5.0 very soon), `exts_default_options` will be ignored because it's not a known easyconfig parameter yet in EasyBuild v3.4.1 & earlier, and so extensions will be downloaded via the 'parent' `source_urls`.

Reply:
Installing the development version did not work in time. I will close this PR and try again after 3.5.0 is released.